### PR TITLE
tests: fix unit tests' file path compatibility and error message assertion across Windows and Linux platforms

### PR DESCRIPTION
**Pull request by mayankshah1607**
Signed-off-by: Mayank Shah <mayankshah1614@gmail.com>

**What type of PR is this?**
/kind test

**What this PR does / why we need it**:
fix unit tests' file path compatibility and error message assertion across Windows and Linux platforms

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/azurefile/nodeserver_test.go [hunk 1]
```diff
@@ -128,7 +145,11 @@ func TestNodePublishVolume(t *testing.T) {
 				TargetPath:        targetTest,
 				StagingTargetPath: errorMountSource,
 				Readonly:          true},
-			expectedErr: status.Errorf(codes.Internal, "Could not mount \"./error_mount_source\" at \"./target_test\": fake Mount: source error"),
+			expectedErr: testutil.TestError{
+				DefaultError: status.Errorf(codes.Internal, fmt.Sprintf("Could not mount \"%s\" at \"%s\": fake Mount: source error", errorMountSource, targetTest)),
+				// todo: Not a desired error. This will need a better fix
+				WindowsError: fmt.Errorf("prepare publish failed for %s with error: could not cast to csi proxy class", targetTest),
```
Comment (mayankshah1607) [inline pkg/azurefile/nodeserver_test.go:151]: We're getting this unexpected error in a lot of places while testing on Windows. Will open a follow-up issue to fix this.

Comment (andyzhangx) [reply pkg/azurefile/nodeserver_test.go:151]: I think csi-proxy.exe should run first
refer to https://github.com/kubernetes-csi/csi-proxy/pull/69#discussion_r487687292

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve
let's merge this PR first, thanks.

**Pull request merged**